vioquine/assignment4/CountingIslands.java

```diff
@@ -0,0 +1,83 @@
+import java.util.ArrayList;
+import java.util.List;
+
+class CountingIslands {
+
+    /**
+     * Counts the islands in a given grid
+     *
+     * @param rows    number of rows
+     * @param columns number of columns
+     * @param tiles   grid with land(true) and see(false)
+     * @return number of islands in the grid
+     */
+    static int countIslands(int rows, int columns, boolean[][] tiles) {
```

Comment:
You pass the number of rows and columns to the method but then in markIsland you still just check the length instead. I'd say decide on one of the ways: check the lengths when you need them or keep passing them as params (both are fine by me) and be consistent across your program.